hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/listener/ClientListenerServiceImpl.java

```diff
@@ -307,39 +306,33 @@ private Boolean deregisterListenerInternal(@Nullable UUID userRegistrationId) {
         //This method should only be called from registrationExecutor
         assert (Thread.currentThread().getName().contains("eventRegistration"));
 
-        ClientListenerRegistration listenerRegistration = registrations.get(userRegistrationId);
+        ClientListenerRegistration listenerRegistration = registrations.remove(userRegistrationId);
         if (listenerRegistration == null) {
             return false;
         }
-        boolean successful = true;
 
         Map<Connection, ClientConnectionRegistration> registrations = listenerRegistration.getConnectionRegistrations();
-        Iterator<Map.Entry<Connection, ClientConnectionRegistration>> iterator = registrations.entrySet().iterator();
-        while (iterator.hasNext()) {
-            Map.Entry<Connection, ClientConnectionRegistration> entry = iterator.next();
+        for (Map.Entry<Connection, ClientConnectionRegistration> entry : registrations.entrySet()) {
             ClientConnectionRegistration registration = entry.getValue();
-            Connection subscriber = entry.getKey();
-            try {
-                ListenerMessageCodec listenerMessageCodec = listenerRegistration.getCodec();
-                UUID serverRegistrationId = registration.getServerRegistrationId();
-                ClientMessage request = listenerMessageCodec.encodeRemoveRequest(serverRegistrationId);
-                if (request != null) {
-                    new ClientInvocation(client, request, null, subscriber).invoke().get();
-                }
-                ((ClientConnection) subscriber).removeEventHandler(registration.getCallId());
-                iterator.remove();
-            } catch (Exception e) {
-                if (subscriber.isAlive()) {
-                    successful = false;
-                    logger.warning("Deregistration of listener with ID " + userRegistrationId
-                            + " has failed to address " + subscriber.getRemoteAddress(), e);
-                }
+            ClientConnection subscriber = (ClientConnection) entry.getKey();
+            //remove local handler
+            subscriber.removeEventHandler(registration.getCallId());
+            //the rest is for deleting remote registration
+            ListenerMessageCodec listenerMessageCodec = listenerRegistration.getCodec();
+            UUID serverRegistrationId = registration.getServerRegistrationId();
+            ClientMessage request = listenerMessageCodec.encodeRemoveRequest(serverRegistrationId);
+            if (request == null) {
+                continue;
             }
+            ClientInvocation clientInvocation = new ClientInvocation(client, request, null, subscriber);
+            clientInvocation.setInvocationTimeoutMillis(Long.MAX_VALUE);
+            clientInvocation.invokeUrgent().exceptionally(throwable -> {
+                logger.warning("Deregistration of listener with ID " + userRegistrationId
```

Comment:
Question: what if the underlying connections gets closed due to missed heartbeats or the client shuts down? Shouldn't we avoid log messages in this case (i.e. filter out `HazelcastClientNotActiveException` and `TargetDisconnectedException`)?